components/status/UploadFilesWidget.vue

```diff
@@ -0,0 +1,174 @@
+      Successfully&nbsp;<wbr />Uploaded:
+    </span>
+    <!-- original mockflow ID: cmpD58c69d7de8aa0934dca9ef4e8beabbdc -->
+    <span class="span__upload-file-count-container">
+      {{ value }}/{{ max }}
+    </span>
+    <b-progress
+      id="upload-progress-bar"
+      :value="value"
+      :max="max"
+      variant="success"
+    ></b-progress>
+  </div>
+</template>
+<script>
+import Vue from "vue";
+import { mapState } from "vuex";
+import { BProgress } from "bootstrap-vue";
+Vue.component("BProgress", BProgress);
+
+export default {
+  name: "UploadFilesWidget",
+  data() {
+    return {
+      tick: false,
+    };
+  },
+  computed: {
+    ...mapState("settings", {
+      value: "file_count",
+    }),
+    ...mapState("settings", {
+      max: "max_file_count",
+    }),
+  },
+  watch: {
+    value() {
+      if (this.value != 0) {
+        if (this.max == this.value) {
+          this.tick = true;
+          setTimeout(
+            function () {
+              this.tick = false;
+            }.bind(this),
+            1500
+          );
+        } else {
+          this.tick = false;
+        }
+      }
+    },
+  },
+};
+</script>
+<style>
+.div__upload-file-background {
+  transform: rotate(0deg);
+  box-sizing: border-box;
+  padding: 0px;
+  margin: 0px;
+  background: rgb(17, 17, 17);
+  position: absolute;
+  width: 400px;
+  height: 45px;
+  top: 0px;
+  left: 0px;
+  visibility: visible;
+  border: 0px none rgb(0, 0, 0);
+  border-radius: 0px;
+  box-shadow: none;
+  z-index: 1;
+  pointer-events: all;
+}
+
+.div__upload-custom-check-mark {
+  position: absolute;
+  width: 25px;
+  height: 25px;
+  top: 5px;
+  left: 33px;
+  color: #00c46f;
+  z-index: 10;
+}
+
+.span__upload-file-label-txt {
+  pointer-events: all;
+  line-height: 100%;
+  transform: rotate(0deg);
+  overflow: hidden;
+  position: absolute;
+  width: 170px;
+  height: 30px;
+  top: 9px;
+  left: 50.3571px;
+  padding: 5px;
+  visibility: visible;
+  user-select: none;
+  font-family: Muli;
+  font-weight: normal;
+  font-style: normal;
+  text-decoration: none;
+  font-size: 15px;
+  color: rgb(255, 255, 255);
+  text-align: right;
+  z-index: 3;
+}
+
+.span__upload-file-count-container {
+  pointer-events: all;
+  line-height: 100%;
+  transform: rotate(0deg);
+  overflow: hidden;
+  position: absolute;
+  width: 84px;
+  height: 30px;
+  top: 9.6663px;
+  left: 200.909px;
+  padding: 5px;
+  visibility: visible;
+  user-select: none;
+  font-family: "Anonymous Pro";
+  font-weight: normal;
+  font-style: normal;
+  text-decoration: none;
+  font-size: 15px;
+  color: rgb(183, 183, 183);
+  text-align: right;
+  z-index: 9;
+}
+
+.progress {
+  background-color: #4c4c4c;
+}
+
+#upload-progress-bar {
+  position: absolute;
+  width: 110px;
+  height: 4px;
+  top: 20px;
+  left: 281px;
+  visibility: visible;
+  background-color: #4c4c4c;
+  z-index: 7;
+}
+
+#upload-progress-bar > .bg-success {
+  background-color: #00c46f !important;
```

Comment:
is the `!important` necessary here? is there another way to achieve this without using `!important`?  The Mozilla documentation and other places caution that `!important` should be very rarely used and you should try your best to use the CSS inheritence rules instead

Reply:
/mantarray-frontend-components/node_modules/bootstrap/dist/css/bootstrap.css 

Has CSS property defined in the following format.
```
.bg-success {
  background-color: #28a745 !important;
}
```

In order to replace the progress-bar  background-color, as per MockflowUI, it has to be  as implemented as presently done.

```
#upload-progress-bar > .bg-success {
  background-color: #00c46f !important;
}
```

Currently this works and I don't find any issue, if this has to be very rarely used as per Mozilla documentation, then bootstrap a very popular CSS library is using !important in several part of CSS definition, why didn't the open source, popular library don't follow the inheritance rules of Mozilla documentation ?

Reply:
I don't know why Bootstrap is not following the Mozilla documentation.  Yesterday we agreed it was OK to use the `!important` flag in this case in our code, but please put a comment in the code describing the justification for not following the general Mozilla guidelines

Reply:
Included a comment with the reason for having !important.